internal/core/algorithm/ngt/ngt.go

```diff
@@ -345,16 +345,18 @@ func (n *ngt) Search(vec []float32, size int, epsilon, radius float32) ([]Search
 		n.mu.RUnlock()
 		return nil, n.newGoError(ne)
 	}
-
 	n.mu.RUnlock()
 
-	rsize := int(C.ngt_get_result_size(results, n.ebuf))
-	if rsize == -1 {
+	switch rsize := int(C.ngt_get_result_size(results, n.ebuf)); rsize {
+	case -1:
 		return nil, n.newGoError(n.ebuf)
+	case 0:
+		return nil, errors.ErrEmptySearchResult
```

Comment:
Please add a new test case that it correctly returns `ErrEmptySearchResult` when there's no index.